Adds some documentation to species.dm

trollbreeder opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds documentation to all vars and 40-50% procs in species.dm
Also updates some of the code to use defines instead of 0 and 1

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
DOCUMENTATION = GOODE
HAVING CODE EXPLAINED = GOODE

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

trollbreeder committed:
Merge pull request #1 from tgstation/master

Updates master from source repo

trollbreeder committed:
Documentation on species vars, and roughly 40-50% of procs

I can do no more of this. My mind is dead from boredom.

trollbreeder committed:
thought this proc was deprecated, apparantly its not

cloning removal has permanentally scarred me

trollbreeder commented:
SHIT HOLD ON I SCUFFED ONE SECOND

trollbreeder commented:
Panic over

trollbreeder commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1,58 +1,110 @@
+	var/list/mutant_organs = list()
+	///This affects the race's speed. Positive numbers make it move slower, negative numbers make it move faster.
+	var/speedmod = 0
+	///Overall defense for the race, or less defense, if it's negative.
```
it's clear that the damage mods say multiplier for X damage, but for armor and speed this isn't so obvious, i'll change it

trollbreeder committed:
Updates speed and armor variable docs

It wasn't made clear that armor was percentage based and speed was multiplicative in the documentation. This is now fixed.
Also removes some whitespace.

nemvar commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1,58 +1,110 @@
+	var/default_color = "#FFF"
+
+	///Whether or not the race has sexual characteristics (biological genders). At the moment this is only 0 for skeletons and shadows
+	var/sexes = 1
```
```suggestion
	///Whether or not the race has sexual characteristics (biological genders). At the moment this is only FALSE for skeletons and shadows
	var/sexes = TRUE
```

nemvar commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1,58 +1,110 @@
+	var/hair_alpha = 255
+
+	///Does the species use skintones or not? As of now only used by humans.
+	var/use_skintones = 0
```
```suggestion
	var/use_skintones = FALSE
```

nemvar commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1,58 +1,110 @@
+	/////Inventory slots the race can't equip stuff to. Golems cannot wear jumpsuits, for example.
+	var/list/no_equip = list()
+	/// Allows the species to equip items that normally require a jumpsuit without having one equipped. Used by golems.
+	var/nojumpsuit = 0
```
```suggestion
	var/nojumpsuit = FALSE
```

nemvar reviewed (commented):
I highlighted some things that are booleans but are not using the defines.
Would be nice if you also updated them to current standards.

trollbreeder committed:
Update code/modules/mob/living/carbon/human/species.dm

Co-Authored-By: nemvar <47324920+nemvar@users.noreply.github.com>

trollbreeder committed:
Update code/modules/mob/living/carbon/human/species.dm

Co-Authored-By: nemvar <47324920+nemvar@users.noreply.github.com>

trollbreeder committed:
Update code/modules/mob/living/carbon/human/species.dm

Co-Authored-By: nemvar <47324920+nemvar@users.noreply.github.com>

trollbreeder commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -120,11 +189,26 @@ GLOBAL_LIST_EMPTY(roundstart_races)
+  * This generates a random unique name based on a human's species and gender.
+  * Arguments:
+  * gender - The gender that the name should adhere to. Use MALE for male names, use anything else for female names.
+  * unique -
```
Note to self: Find out what unique does and add it to this documentation

Qustinnus replied:
It tries to ensure that the name is unique and no dupes exist in the station rn

cacogen commented:
egood

Qustinnus commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1,58 +1,110 @@
+	var/meat = /obj/item/reagent_containers/food/snacks/meat/slab/human
+	///What skin the species drops when gibbed by a gibber machine.
 	var/skinned_type
+	///Food type that the species likes, giving them a mood boost. Lizards like meat, for example.
```
pluralize type to types else people might not realize its a bitfield, on that note, maybe mention its a bitfield.

Qustinnus commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1,58 +1,110 @@
 	var/liked_food = NONE
+	///Food type that the species dislikes, giving them disgust. Humans hate raw food, for example.
 	var/disliked_food = GROSS
+	///Food type that is toxic to the species, giving them toxin damage. Meat is toxic to moths, for example.
```
this is not what toxic category food does; it's a shit name (which is my bad) but its generally closed to hated foods, the toxic damage is usually the reagents themselves. It just raises disgust more than disliked food

Qustinnus commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1,58 +1,110 @@
-	var/exotic_blood = ""	// If your race wants to bleed something other than bog standard blood, change this to reagent id.
-	var/exotic_bloodtype = "" //If your race uses a non standard bloodtype (A+, O-, AB-, etc)
-	var/meat = /obj/item/reagent_containers/food/snacks/meat/slab/human //What the species drops on gibbing
+	///This allows races to have specific hair colors. If null, it uses the H's hair/facial hair colors. If "mutcolor", it uses the H's mutant_color.
```
I forgot to add this to this comment but "fixedmutcolor" makes it use fixedmutcolor.

Qustinnus commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1,58 +1,110 @@
+	///Base electrocution coefficient.  Basically a multiplier for damage from electrocutions.
+	var/siemens_coeff = 1
+	///What kind of damage overlays (if any) appear on our species when wounded?
+	var/damage_overlay_type = "human"
```
mention that "" means no overlays.

trollbreeder committed:
Documentation updates with Qustinuss

Mentions fixedmutcolor in hair_color
Makes preferred food_type vars more clear that they are bitfields.
toxic_food is not toxic, it's just more disgusting than disliked food
Updates overlay var doc to mention that an empty string adds no damage overlays.

trollbreeder committed:
Defines uniqueness, also lastname docs change

Updates generate name proc to have info on the unique argument
Changes override this to set to false on lastname. Overriding is done with procs, not vars or arguments

TiviPlus commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -142,11 +226,24 @@ GLOBAL_LIST_EMPTY(roundstart_races)
-//Called when cloning, copies some vars that should be kept
+/**
+  * Called when cloning, copies some vars that should be kept
+  *
```
```suggestion
/**
```
Cloning is kill

trollbreeder replied:
Yeah but this is still used by some species, like slime dudes

TiviPlus replied:
Yes leave that bit I'm just saying to remove the reference to cloning

trollbreeder replied:
cloning is what i'd best describe what this proc does even if cloning is kill

TiviPlus replied:
"Called when cloning" describes when the proc is called and not what it does, so at least remove that bit, though "copies some vars that should be kept" can be left in

trollbreeder replied:
i guess that makes sense, tivi

TiviPlus commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -120,11 +189,26 @@ GLOBAL_LIST_EMPTY(roundstart_races)
+  * Arguments:
+  * gender - The gender that the name should adhere to. Use MALE for male names, use anything else for female names.
+  * unique - If true, ensures that this new name is not a duplicate of anyone else's name currently on the station.
+  * lastname - Does this species' naming system adhere to the last name system? Set to false if it doesn't.
```
Be a dear please and make the arguments use the format specified in the autodoc guide(* * arg1 - why is this relevant)

trollbreeder replied:
They're in the order from arg1 to arg3, and i named them after the variables that they're used in the proc. I think arg1 is simply a placeholder for the argument's name.

TiviPlus replied:
![Capture](https://user-images.githubusercontent.com/57223640/77250196-84d0ac00-6c46-11ea-862c-d45aaf1aaf9c.PNG)
You need two asterisks for arguments, see example

trollbreeder replied:
i'll do that, even though regenerate_organs doesn't do this

trollbreeder committed:
updates proc docs to use double asterisk

if this fails travis documentation i'm gonna be angery

trollbreeder committed:
i guess that makes sense

updates copy properties proc with better docs

trollbreeder committed:
Part 2 of i guess this makes sense

i should stop early commiting y'know

Rohesie commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1,58 +1,110 @@
-	var/list/no_equip = list()	// slots the race can't equip stuff to
-	var/nojumpsuit = 0	// this is sorta... weird. it basically lets you equip stuff that usually needs jumpsuits without one, like belts and pockets and ids
-	var/say_mod = "says"	// affects the speech message
+	/////Inventory slots the race can't equip stuff to. Golems cannot wear jumpsuits, for example.
```
```suggestion
	///Inventory slots the race can't equip stuff to. Golems cannot wear jumpsuits, for example.
```

trollbreeder committed:
Update code/modules/mob/living/carbon/human/species.dm

Co-Authored-By: Rohesie <rohesie@gmail.com>
